src/test/regress/output/tablespace.source

```diff
@@ -251,6 +251,29 @@ ALTER TABLE ALL IN TABLESPACE testspace_renamed SET TABLESPACE pg_default;
 NOTICE:  no matching relations in tablespace "testspace_renamed" found
 -- Should succeed
 DROP TABLESPACE testspace_renamed;
+-- Create a new tablespace and check that within this tablespace,
+-- after dropping the table, the size of all deleted (but still opened) files is 0.
+CREATE TABLESPACE testspace LOCATION '@testtablespace@';
+DROP TABLE IF EXISTS t;
+-- Checking that the AO row-oriented table doesn't leave non-zero files.
+CREATE TABLE t(a int, b int) WITH (appendoptimized = true, orientation=row) TABLESPACE testspace;
+INSERT INTO t SELECT i, i FROM generate_series(1,1000) i;
+DROP TABLE t;
+\! find /proc/[0-9]*/fd -lname '@testtablespace@*(deleted)' -exec stat -Lc '%s' {} \; 2>/dev/null | awk 'BEGIN {sum=0} {sum += $1} END {print "size:",  sum}';
+size: 0
+-- Checking that the AO column-oriented table doesn't leave non-zero files.
+CREATE TABLE t(a int, b int) WITH (appendoptimized = true, orientation=column) TABLESPACE testspace;
+INSERT INTO t SELECT i, i FROM generate_series(1,1000) i;
+DROP TABLE t;
+\! find /proc/[0-9]*/fd -lname '@testtablespace@*(deleted)' -exec stat -Lc '%s' {} \; 2>/dev/null | awk 'BEGIN {sum=0} {sum += $1} END {print "size:",  sum}';
+size: 0
+-- Checking that the temp heap table doesn't leave non-zero files.
+CREATE TEMP TABLE t(a int, b int) TABLESPACE testspace;
+INSERT INTO t SELECT i, i FROM generate_series(1,20000000) i;
```

Comment:
>I was able to reproduce it in the auto test too, by commands:... but it requires more data to add to the table (it is not reproducable on my side if I insert significantly less data). If everybody is Ok with such test case, I will add it to PR.

1) 20000000 is too much, the number of tuples should be as small as possible for the regress suite. Otherwise, I suggest not to include the test.

Reply:
Ok, I will spend some time to investigate if we can reproduce the problem with less data inserted.

Reply:
Test case was updated.